diff --git a/CMakeLists.txt b/CMakeLists.txt
index 992e661..74a69d6 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,8 @@ PROJECT (TMC2 CXX C )
 SET( TMC2_VERSION_MAJOR 15 )
 SET( TMC2_VERSION_MINOR 0  )
 
+SET(TMC2_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
 # Video encoder and decoder library and application:
 OPTION( USE_JMAPP_VIDEO_CODEC     "Use JM application to encode and decode video"           TRUE  )  # JMAPP  = 0
 OPTION( USE_HMAPP_VIDEO_CODEC     "Use HM application to encode and decode video"           TRUE  )  # HMAPP  = 1
@@ -26,7 +28,7 @@ OPTION( ENABLE_PAPI_PROFILING     "Enable PAPI profiling"
 set( CMAKE_CXX_STANDARD            14 )
 set( CMAKE_CXX_STANDARD_REQUIRED   ON )
 set( CMAKE_EXPORT_COMPILE_COMMANDS ON )
-set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/dependencies/cmake")
+set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${TMC2_SOURCE_DIR}/dependencies/cmake")
 
 INCLUDE( CheckCXXCompilerFlag )
 CHECK_CXX_COMPILER_FLAG( "-std=c++14" COMPILER_SUPPORTS_CXX14 )
@@ -41,13 +43,13 @@ ELSE ()
 ENDIF()
 
 IF( ENABLE_PAPI_PROFILING )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/papi.cmake ) 
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/papi.cmake )
 ENDIF()
 # SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-function -Wno-implicit -Wno-sign-compare -Wall -Wformat=0" )
 
-SET( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/lib )
-SET( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/lib )
-SET( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin )
+SET( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${TMC2_SOURCE_DIR}/lib )
+SET( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${TMC2_SOURCE_DIR}/lib )
+SET( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${TMC2_SOURCE_DIR}/bin )
 IF (MSVC)
   MESSAGE(STATUS "MSVC OPTIONS")
   ADD_DEFINITIONS("/wd4244 /wd4267 /wd4996")
@@ -57,25 +59,25 @@ ENDIF()
 
 # Include external libraries
 IF( USE_HMLIB_VIDEO_CODEC )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/hm.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/hm.cmake )
 ENDIF()
 IF( USE_JMLIB_VIDEO_CODEC )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/jm_lib.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/jm_lib.cmake )
 ENDIF()
 IF( USE_JMAPP_VIDEO_CODEC )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/jm_app.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/jm_app.cmake )
 ENDIF()
 IF( USE_FFMPEG_VIDEO_CODEC )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/ffmpeg.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/ffmpeg.cmake )
 ENDIF()
 IF( USE_VTMLIB_VIDEO_CODEC )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/vtm.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/vtm.cmake )
 ENDIF()
 IF( USE_SHMAPP_VIDEO_CODEC )
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/shm_app.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/shm_app.cmake )
 ENDIF()
 IF( USE_HDRTOOLS )  
-  INCLUDE( ${CMAKE_SOURCE_DIR}/dependencies/cmake/hdrtools.cmake )
+  INCLUDE( ${TMC2_SOURCE_DIR}/dependencies/cmake/hdrtools.cmake )
 ENDIF()
 
 ## CLANG FORMAT
@@ -94,12 +96,12 @@ ADD_SUBDIRECTORY(source/lib/PccLibEncoder)
 ADD_SUBDIRECTORY(source/lib/PccLibMetrics)
 ADD_SUBDIRECTORY(source/lib/PccLibConformance)
 
-ADD_SUBDIRECTORY(source/app/PccAppParser)
-ADD_SUBDIRECTORY(source/app/PccAppEncoder)
-ADD_SUBDIRECTORY(source/app/PccAppDecoder)
-ADD_SUBDIRECTORY(source/app/PccAppMetrics)
-ADD_SUBDIRECTORY(source/app/PccAppConformance)
-ADD_SUBDIRECTORY(source/app/PccAppVideoEncoder)
-ADD_SUBDIRECTORY(source/app/PccAppVideoDecoder)
-ADD_SUBDIRECTORY(source/app/PccAppColorConverter)
-ADD_SUBDIRECTORY(source/app/PccAppNormalGenerator)
+#ADD_SUBDIRECTORY(source/app/PccAppParser)
+#ADD_SUBDIRECTORY(source/app/PccAppEncoder)
+#ADD_SUBDIRECTORY(source/app/PccAppDecoder)
+#ADD_SUBDIRECTORY(source/app/PccAppMetrics)
+#ADD_SUBDIRECTORY(source/app/PccAppConformance)
+#ADD_SUBDIRECTORY(source/app/PccAppVideoEncoder)
+#ADD_SUBDIRECTORY(source/app/PccAppVideoDecoder)
+#ADD_SUBDIRECTORY(source/app/PccAppColorConverter)
+#ADD_SUBDIRECTORY(source/app/PccAppNormalGenerator)
diff --git a/dependencies/cmake/hdrtools.cmake b/dependencies/cmake/hdrtools.cmake
index a49f7a2..8604dcf 100755
--- a/dependencies/cmake/hdrtools.cmake
+++ b/dependencies/cmake/hdrtools.cmake
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
 
   
-SET( HDRTOOLS_DIR             ${CMAKE_SOURCE_DIR}/dependencies/HDRTools )
+SET( HDRTOOLS_DIR             ${TMC2_SOURCE_DIR}/dependencies/HDRTools )
 MESSAGE("Clone and build HDRTools libraries: ${HDRTOOLS_DIR}") 
 
 IF( NOT EXISTS "${HDRTOOLS_DIR}/CMakeLists.txt" )   
diff --git a/dependencies/cmake/hm.cmake b/dependencies/cmake/hm.cmake
index 942c514..2932da1 100755
--- a/dependencies/cmake/hm.cmake
+++ b/dependencies/cmake/hm.cmake
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
 
 SET( HM_VERSION         HM-16.20+SCM-8.8 )
-SET( HM_DIR             ${CMAKE_SOURCE_DIR}/dependencies/${HM_VERSION}/ )
+SET( HM_DIR             ${TMC2_SOURCE_DIR}/dependencies/${HM_VERSION}/ )
 set( HM_LIB_SOURCE_DIR  ${HM_DIR}/source/Lib )
 MESSAGE("Clone and build HM libraries: ${HM_LIB_SOURCE_DIR}") 
 
@@ -17,9 +17,9 @@ ENDIF()
 
 IF( NOT EXISTS "${HM_DIR}/PATCHED" )
   IF ( USE_HM_PCC_RDO )
-    SET( HM_PATCH ${CMAKE_SOURCE_DIR}/dependencies/hm-modification/HM-16.20+SCM-8.8_with_RDO.patch )
+    SET( HM_PATCH ${TMC2_SOURCE_DIR}/dependencies/hm-modification/HM-16.20+SCM-8.8_with_RDO.patch )
   ELSE()
-    SET( HM_PATCH ${CMAKE_SOURCE_DIR}/dependencies/hm-modification/HM-16.20+SCM-8.8.patch )
+    SET( HM_PATCH ${TMC2_SOURCE_DIR}/dependencies/hm-modification/HM-16.20+SCM-8.8.patch )
   ENDIF()
   MESSAGE("HM patch: ${HM_DIR} (${HM_PATCH})")
   EXECUTE_PROCESS( COMMAND git apply ${HM_PATCH} --whitespace=nowarn WORKING_DIRECTORY ${HM_DIR} RESULT_VARIABLE ret )
diff --git a/dependencies/cmake/jm_app.cmake b/dependencies/cmake/jm_app.cmake
index 82c5378..ad45565 100644
--- a/dependencies/cmake/jm_app.cmake
+++ b/dependencies/cmake/jm_app.cmake
@@ -1,11 +1,11 @@
 cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
 
 SET( JM_VERSION         jm19.0_app )
-SET( JM_DIR             ${CMAKE_SOURCE_DIR}/dependencies/${JM_VERSION}/ )
+SET( JM_DIR             ${TMC2_SOURCE_DIR}/dependencies/${JM_VERSION}/ )
 SET( JM_APP_SOURCE_DIR  ${JM_DIR}/source )
 MESSAGE("Clone the JM source code: ${JM_DIR}") 
   
 IF( NOT EXISTS "${JM_DIR}/CMakeLists.txt" )   
   EXECUTE_PROCESS( COMMAND git -c http.sslVerify=false clone --depth 1 https://vcgit.hhi.fraunhofer.de/jct-vc/JM.git ${JM_DIR} )
-  #EXECUTE_PROCESS( COMMAND git apply ${CMAKE_SOURCE_DIR}/dependencies/jm-modification/SampleStreamITRI.patch WORKING_DIRECTORY ${JM_DIR} )    
+  #EXECUTE_PROCESS( COMMAND git apply ${TMC2_SOURCE_DIR}/dependencies/jm-modification/SampleStreamITRI.patch WORKING_DIRECTORY ${JM_DIR} )
 ENDIF()
diff --git a/dependencies/cmake/jm_lib.cmake b/dependencies/cmake/jm_lib.cmake
index a2c9cdd..00eb4a0 100644
--- a/dependencies/cmake/jm_lib.cmake
+++ b/dependencies/cmake/jm_lib.cmake
@@ -1,14 +1,14 @@
 cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
 
 SET( JM_VERSION         jm19.0_lib )
-SET( JM_DIR             ${CMAKE_SOURCE_DIR}/dependencies/${JM_VERSION}/ )
+SET( JM_DIR             ${TMC2_SOURCE_DIR}/dependencies/${JM_VERSION}/ )
 SET( JM_LIB_SOURCE_DIR  ${JM_DIR}/source )
 MESSAGE("Clone and build JM libraries for JM LIB: ${JM_DIR}") 
   
 IF( NOT EXISTS "${JM_DIR}/CMakeLists.txt" )   
   EXECUTE_PROCESS( COMMAND git -c http.sslVerify=false clone --depth 1 https://vcgit.hhi.fraunhofer.de/jct-vc/JM.git ${JM_DIR} ) 
-  EXECUTE_PROCESS( COMMAND git apply ${CMAKE_SOURCE_DIR}/dependencies/jm-modification/PCC_JM.patch WORKING_DIRECTORY ${JM_DIR} )
-  #EXECUTE_PROCESS( COMMAND git apply ${CMAKE_SOURCE_DIR}/dependencies/jm-modification/SampleStreamITRI.patch WORKING_DIRECTORY ${JM_DIR} )  
+  EXECUTE_PROCESS( COMMAND git apply ${TMC2_SOURCE_DIR}/dependencies/jm-modification/PCC_JM.patch WORKING_DIRECTORY ${JM_DIR} )
+  #EXECUTE_PROCESS( COMMAND git apply ${TMC2_SOURCE_DIR}/dependencies/jm-modification/SampleStreamITRI.patch WORKING_DIRECTORY ${JM_DIR} )
 ENDIF()
 
 MESSAGE("JM_DIR       : ${JM_DIR}")
diff --git a/dependencies/cmake/papi.cmake b/dependencies/cmake/papi.cmake
index 3bb9b3d..1f6aac5 100644
--- a/dependencies/cmake/papi.cmake
+++ b/dependencies/cmake/papi.cmake
@@ -7,13 +7,13 @@ ENDIF()
 
 IF( ENABLE_PAPI_PROFILING )
   MESSAGE( "PAPI profiling enable (ENABLE_PAPI_PROFILING = " ${ENABLE_PAPI_PROFILING} " ).")
-  IF( NOT IS_DIRECTORY "${CMAKE_SOURCE_DIR}/dependencies/papi/" )    
-    EXECUTE_PROCESS( COMMAND git clone --depth 1 https://bitbucket.org/icl/papi.git ${CMAKE_SOURCE_DIR}/dependencies/papi/; )    
+  IF( NOT IS_DIRECTORY "${TMC2_SOURCE_DIR}/dependencies/papi/" )
+    EXECUTE_PROCESS( COMMAND git clone --depth 1 https://bitbucket.org/icl/papi.git ${TMC2_SOURCE_DIR}/dependencies/papi/; )
   ENDIF()
-  IF ( NOT EXISTS ${CMAKE_SOURCE_DIR}/dependencies/papi/src/libpapi.a )
-    EXECUTE_PROCESS( COMMAND ./configure WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/dependencies/papi/src/ )
-    EXECUTE_PROCESS( COMMAND make -j4    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/dependencies/papi/src/ )
+  IF ( NOT EXISTS ${TMC2_SOURCE_DIR}/dependencies/papi/src/libpapi.a )
+    EXECUTE_PROCESS( COMMAND ./configure WORKING_DIRECTORY ${TMC2_SOURCE_DIR}/dependencies/papi/src/ )
+    EXECUTE_PROCESS( COMMAND make -j4    WORKING_DIRECTORY ${TMC2_SOURCE_DIR}/dependencies/papi/src/ )
   ENDIF()
-  SET( CMAKE_CXX_FLAGS        "${CMAKE_CXX_FLAGS} -I${CMAKE_SOURCE_DIR}/dependencies/papi/src/ " )
-  SET( CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CMAKE_SOURCE_DIR}/dependencies/papi/src/libpapi.a ")
+  SET( CMAKE_CXX_FLAGS        "${CMAKE_CXX_FLAGS} -I${TMC2_SOURCE_DIR}/dependencies/papi/src/ " )
+  SET( CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${TMC2_SOURCE_DIR}/dependencies/papi/src/libpapi.a ")
 ENDIF()
diff --git a/dependencies/cmake/shm_app.cmake b/dependencies/cmake/shm_app.cmake
index 3b54d4f..799eeb6 100644
--- a/dependencies/cmake/shm_app.cmake
+++ b/dependencies/cmake/shm_app.cmake
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
 
 SET( SHM_VERSION         SHM-12.4 )
-SET( SHM_DIR             ${CMAKE_SOURCE_DIR}/dependencies/${SHM_VERSION}/ )
+SET( SHM_DIR             ${TMC2_SOURCE_DIR}/dependencies/${SHM_VERSION}/ )
 set( SHM_APP_SOURCE_DIR  ${SHM_DIR}/source )
 MESSAGE("Clone the SHM source code: ${SHM_DIR}") 
 
diff --git a/dependencies/cmake/vtm.cmake b/dependencies/cmake/vtm.cmake
index ba66e10..addee47 100755
--- a/dependencies/cmake/vtm.cmake
+++ b/dependencies/cmake/vtm.cmake
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
 SET( VTM_VERSION         VTM-13.0 )
-SET( VTM_DIR             ${CMAKE_SOURCE_DIR}/dependencies/VTM )
+SET( VTM_DIR             ${TMC2_SOURCE_DIR}/dependencies/VTM )
 set( VTM_LIB_SOURCE_DIR  ${VTM_DIR}/source/Lib )
 MESSAGE("Clone and build VTM libraries: ${VTM_LIB_SOURCE_DIR}") 
 
@@ -17,7 +17,7 @@ ENDIF()
 
 IF( NOT EXISTS "${VTM_DIR}/PATCHED" )
   MESSAGE("VTM patch: ${VTM_DIR}")
-  SET( VTM_PATCH ${CMAKE_SOURCE_DIR}/dependencies/vtm-modification/adaptions_for_vtm_13_0.patch )
+  SET( VTM_PATCH ${TMC2_SOURCE_DIR}/dependencies/vtm-modification/adaptions_for_vtm_13_0.patch )
   EXECUTE_PROCESS( COMMAND git apply ${VTM_PATCH} --whitespace=nowarn WORKING_DIRECTORY ${VTM_DIR} RESULT_VARIABLE ret )
   IF( NOT ${ret} EQUAL "0")
     MESSAGE( FATAL_ERROR "Error during the VTM patch process. Check that git is well installed on your system." )
diff --git a/dependencies/tbb/Makefile b/dependencies/tbb/Makefile
index 9709201..a6d7764 100644
--- a/dependencies/tbb/Makefile
+++ b/dependencies/tbb/Makefile
@@ -42,7 +42,7 @@ EQUALS = =
 CMAKE_EDIT_COMMAND = /usr/bin/ccmake
 
 # The top-level source directory on which CMake was run.
-CMAKE_SOURCE_DIR = /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress
+TMC2_SOURCE_DIR = /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress
 
 # The top-level build directory on which CMake was run.
 CMAKE_BINARY_DIR = /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress
@@ -53,7 +53,7 @@ CMAKE_BINARY_DIR = /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in
 # Special rule for the target edit_cache
 edit_cache:
 	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running CMake cache editor..."
-	/usr/bin/ccmake -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR)
+	/usr/bin/ccmake -H$(TMC2_SOURCE_DIR) -B$(CMAKE_BINARY_DIR)
 .PHONY : edit_cache
 
 # Special rule for the target edit_cache
@@ -104,7 +104,7 @@ list_install_components/fast: list_install_components
 # Special rule for the target rebuild_cache
 rebuild_cache:
 	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running CMake to regenerate build system..."
-	/usr/bin/cmake -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR)
+	/usr/bin/cmake -H$(TMC2_SOURCE_DIR) -B$(CMAKE_BINARY_DIR)
 .PHONY : rebuild_cache
 
 # Special rule for the target rebuild_cache
@@ -139,7 +139,7 @@ preinstall/fast:
 
 # clear depends
 depend:
-	cd /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress && $(CMAKE_COMMAND) -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR) --check-build-system CMakeFiles/Makefile.cmake 1
+	cd /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress && $(CMAKE_COMMAND) -H$(TMC2_SOURCE_DIR) -B$(CMAKE_BINARY_DIR) --check-build-system CMakeFiles/Makefile.cmake 1
 .PHONY : depend
 
 # Convenience name for target.
@@ -1194,6 +1194,6 @@ help:
 # No rule that depends on this can have commands that come from listfiles
 # because they might be regenerated.
 cmake_check_build_system:
-	cd /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress && $(CMAKE_COMMAND) -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR) --check-build-system CMakeFiles/Makefile.cmake 0
+	cd /home/wp21/PCC/ricardj/gitclones_dev/release-v1.2-proposal-in-progress && $(CMAKE_COMMAND) -H$(TMC2_SOURCE_DIR) -B$(CMAKE_BINARY_DIR) --check-build-system CMakeFiles/Makefile.cmake 0
 .PHONY : cmake_check_build_system
 
diff --git a/source/app/PccAppColorConverter/CMakeLists.txt b/source/app/PccAppColorConverter/CMakeLists.txt
index 9998c39..6bde8dd 100644
--- a/source/app/PccAppColorConverter/CMakeLists.txt
+++ b/source/app/PccAppColorConverter/CMakeLists.txt
@@ -6,12 +6,12 @@ SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
 FILE(GLOB SRC *.h *.cpp *.c 
-                ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* )
+                ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/* )
                      
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include      
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibColorConverter/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite  )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibColorConverter/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite  )
 
 SET( LIBS PccLibCommon PccLibBitstreamCommon PccLibColorConverter ) 
 
diff --git a/source/app/PccAppConformance/CMakeLists.txt b/source/app/PccAppConformance/CMakeLists.txt
index 39e5d1a..74dcef4 100644
--- a/source/app/PccAppConformance/CMakeLists.txt
+++ b/source/app/PccAppConformance/CMakeLists.txt
@@ -6,12 +6,12 @@ SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
 FILE(GLOB SRC *.h *.cpp *.c 
-                ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* )
+                ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/* )
                      
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include      
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibConformance/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite  )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibConformance/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite  )
 
 SET( LIBS PccLibCommon PccLibBitstreamCommon PccLibConformance ) 
 
diff --git a/source/app/PccAppDecoder/CMakeLists.txt b/source/app/PccAppDecoder/CMakeLists.txt
index 6dc131e..56ecaf2 100644
--- a/source/app/PccAppDecoder/CMakeLists.txt
+++ b/source/app/PccAppDecoder/CMakeLists.txt
@@ -5,17 +5,17 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
-FILE(GLOB SRC *.h *.cpp *.c ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* )
-
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibDecoder/include                
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibMetrics/include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibConformance/include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamReader/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibConformance/include )
+FILE(GLOB SRC *.h *.cpp *.c ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/* )
+
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibDecoder/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibMetrics/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibConformance/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamReader/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibConformance/include )
 
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/app/PccAppEncoder/CMakeLists.txt b/source/app/PccAppEncoder/CMakeLists.txt
index f324cec..a743173 100644
--- a/source/app/PccAppEncoder/CMakeLists.txt
+++ b/source/app/PccAppEncoder/CMakeLists.txt
@@ -5,18 +5,18 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
-FILE(GLOB SRC *.h *.cpp *.c ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* 
-                            ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.hpp
-                            ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.h )
+FILE(GLOB SRC *.h *.cpp *.c ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/*
+                            ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.hpp
+                            ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.h )
 
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibEncoder/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibMetrics/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamWriter/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/nanoflann  )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibEncoder/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibMetrics/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamWriter/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/dependencies/nanoflann  )
                      
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/app/PccAppMetrics/CMakeLists.txt b/source/app/PccAppMetrics/CMakeLists.txt
index 89eadcb..2e5b488 100644
--- a/source/app/PccAppMetrics/CMakeLists.txt
+++ b/source/app/PccAppMetrics/CMakeLists.txt
@@ -5,13 +5,13 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
-FILE(GLOB SRC *.h *.cpp *.c ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* )
+FILE(GLOB SRC *.h *.cpp *.c ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/* )
 
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibMetrics/include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibMetrics/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include )
 
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/app/PccAppNormalGenerator/CMakeLists.txt b/source/app/PccAppNormalGenerator/CMakeLists.txt
index 277b9bd..a308c79 100644
--- a/source/app/PccAppNormalGenerator/CMakeLists.txt
+++ b/source/app/PccAppNormalGenerator/CMakeLists.txt
@@ -5,16 +5,16 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
-FILE(GLOB SRC *.h *.cpp *.c ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* 
-                            ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.hpp
-                            ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.h )
+FILE(GLOB SRC *.h *.cpp *.c ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/*
+                            ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.hpp
+                            ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.h )
 
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibEncoder/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/nanoflann  )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibEncoder/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/dependencies/nanoflann  )
                      
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/app/PccAppParser/CMakeLists.txt b/source/app/PccAppParser/CMakeLists.txt
index 81ad53f..43d3888 100644
--- a/source/app/PccAppParser/CMakeLists.txt
+++ b/source/app/PccAppParser/CMakeLists.txt
@@ -7,8 +7,8 @@ PROJECT(${MYNAME} C CXX)
 
 FILE(GLOB SRC *.h *.cpp *.c )
 
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamReader/include )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamReader/include )
 
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/app/PccAppVideoDecoder/CMakeLists.txt b/source/app/PccAppVideoDecoder/CMakeLists.txt
index 3345db2..9015044 100644
--- a/source/app/PccAppVideoDecoder/CMakeLists.txt
+++ b/source/app/PccAppVideoDecoder/CMakeLists.txt
@@ -5,12 +5,12 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
-FILE(GLOB SRC *.h *.cpp *.c ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* )
+FILE(GLOB SRC *.h *.cpp *.c ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/* )
 
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include       
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibVideoDecoder/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite  )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibVideoDecoder/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite  )
                      
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/app/PccAppVideoEncoder/CMakeLists.txt b/source/app/PccAppVideoEncoder/CMakeLists.txt
index 7f99600..b5b8e4a 100644
--- a/source/app/PccAppVideoEncoder/CMakeLists.txt
+++ b/source/app/PccAppVideoEncoder/CMakeLists.txt
@@ -5,12 +5,12 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 SET( MYNAME ${MYNAME}${CMAKE_DEBUG_POSTFIX} )
 PROJECT(${MYNAME} C CXX)
 
-FILE(GLOB SRC *.h *.cpp *.c ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite/* )
+FILE(GLOB SRC *.h *.cpp *.c ${TMC2_SOURCE_DIR}/dependencies/program-options-lite/* )
 
-INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include                    
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include      
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibVideoEncoder/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/program-options-lite  )
+INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibVideoEncoder/include
+                     ${TMC2_SOURCE_DIR}/dependencies/program-options-lite  )
                      
 ADD_EXECUTABLE( ${MYNAME} ${SRC} )
 
diff --git a/source/lib/PccLibBitstreamCommon/CMakeLists.txt b/source/lib/PccLibBitstreamCommon/CMakeLists.txt
index 21ae610..919a087 100644
--- a/source/lib/PccLibBitstreamCommon/CMakeLists.txt
+++ b/source/lib/PccLibBitstreamCommon/CMakeLists.txt
@@ -13,10 +13,10 @@ CONFIGURE_FILE( ${CMAKE_CURRENT_SOURCE_DIR}/include/PCCConfig.h.in
                 ${CMAKE_CURRENT_SOURCE_DIR}/include/PCCConfig.h )
 
 FILE(GLOB PROJECT_INC_FILES include/*.h 
-                ${CMAKE_SOURCE_DIR}/dependencies/libmd5/*.h )
+                ${TMC2_SOURCE_DIR}/dependencies/libmd5/*.h )
     
 FILE(GLOB PROJECT_CPP_FILES source/*.cpp
-                ${CMAKE_SOURCE_DIR}/dependencies/libmd5/*.c )
+                ${TMC2_SOURCE_DIR}/dependencies/libmd5/*.c )
 
 FILE(GLOB PROJECT_IN_FILES  *.in)
 
@@ -25,7 +25,7 @@ SOURCE_GROUP( cpp   FILES ${PROJECT_CPP_FILES} )
 SOURCE_GROUP( input FILES ${PROJECT_IN_FILES}  )
 
 INCLUDE_DIRECTORIES( include
-                ${CMAKE_SOURCE_DIR}/dependencies/libmd5)
+                ${TMC2_SOURCE_DIR}/dependencies/libmd5)
  
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} ${PROJECT_INC_FILES} ${PROJECT_CPP_FILES} ${PROJECT_IN_FILES} )
 
diff --git a/source/lib/PccLibBitstreamReader/CMakeLists.txt b/source/lib/PccLibBitstreamReader/CMakeLists.txt
index c11e653..1daea09 100644
--- a/source/lib/PccLibBitstreamReader/CMakeLists.txt
+++ b/source/lib/PccLibBitstreamReader/CMakeLists.txt
@@ -7,7 +7,7 @@ PROJECT(${MYNAME} C CXX)
 FILE(GLOB SRC include/*.h source/*.cpp )
  
 INCLUDE_DIRECTORIES( include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include )
 
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} )
 
diff --git a/source/lib/PccLibBitstreamWriter/CMakeLists.txt b/source/lib/PccLibBitstreamWriter/CMakeLists.txt
index 96c1385..2f33287 100644
--- a/source/lib/PccLibBitstreamWriter/CMakeLists.txt
+++ b/source/lib/PccLibBitstreamWriter/CMakeLists.txt
@@ -7,7 +7,7 @@ PROJECT(${MYNAME} C CXX)
 FILE(GLOB SRC include/*.h source/*.cpp )
 
 INCLUDE_DIRECTORIES( include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include )
 
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} )
 
diff --git a/source/lib/PccLibColorConverter/CMakeLists.txt b/source/lib/PccLibColorConverter/CMakeLists.txt
index 023b7ba..222e639 100644
--- a/source/lib/PccLibColorConverter/CMakeLists.txt
+++ b/source/lib/PccLibColorConverter/CMakeLists.txt
@@ -10,8 +10,8 @@ FILE(GLOB SRC  include/*.h
                ${HDRTOOLS_DIR}/projects/HDRConvert/src/ProjectParameters.cpp )
 
 INCLUDE_DIRECTORIES( include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include/
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include/
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include/
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include/
                      ${HDRTOOLS_DIR}/common/inc
                      ${HDRTOOLS_DIR}/projects/HDRConvert/inc )
 
diff --git a/source/lib/PccLibCommon/CMakeLists.txt b/source/lib/PccLibCommon/CMakeLists.txt
index 059e95e..8415d2d 100644
--- a/source/lib/PccLibCommon/CMakeLists.txt
+++ b/source/lib/PccLibCommon/CMakeLists.txt
@@ -13,11 +13,11 @@ CONFIGURE_FILE( ${CMAKE_CURRENT_SOURCE_DIR}/include/PCCConfig.h.in
                 ${CMAKE_CURRENT_SOURCE_DIR}/include/PCCConfig.h )
 
 FILE(GLOB PROJECT_INC_FILES include/*.h
-                            ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.hpp
-                            ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.h
-                            ${CMAKE_SOURCE_DIR}/dependencies/libmd5/*.h )
+                            ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.hpp
+                            ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.h
+                            ${TMC2_SOURCE_DIR}/dependencies/libmd5/*.h )
 FILE(GLOB PROJECT_CPP_FILES source/*.cpp 
-                            ${CMAKE_SOURCE_DIR}/dependencies/libmd5/*.c )
+                            ${TMC2_SOURCE_DIR}/dependencies/libmd5/*.c )
 FILE(GLOB PROJECT_IN_FILES  *.in )
 
 SOURCE_GROUP( inc   FILES ${PROJECT_INC_FILES} )
@@ -25,14 +25,14 @@ SOURCE_GROUP( cpp   FILES ${PROJECT_CPP_FILES} )
 SOURCE_GROUP( input FILES ${PROJECT_IN_FILES}  )
 
 INCLUDE_DIRECTORIES( include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/nanoflann
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/libmd5 )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/dependencies/nanoflann
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/dependencies/libmd5 )
  
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} ${PROJECT_INC_FILES} ${PROJECT_CPP_FILES} ${PROJECT_IN_FILES} )
 IF( ENABLE_PAPI_PROFILING )
-  TARGET_LINK_LIBRARIES(${MYNAME} ${CMAKE_SOURCE_DIR}/dependencies/papi/src/libpapi.a  )
+  TARGET_LINK_LIBRARIES(${MYNAME} ${TMC2_SOURCE_DIR}/dependencies/papi/src/libpapi.a  )
 ENDIF()
 
 TARGET_LINK_LIBRARIES(${MYNAME} PccLibBitstreamCommon )
diff --git a/source/lib/PccLibConformance/CMakeLists.txt b/source/lib/PccLibConformance/CMakeLists.txt
index ce18eeb..c688875 100644
--- a/source/lib/PccLibConformance/CMakeLists.txt
+++ b/source/lib/PccLibConformance/CMakeLists.txt
@@ -8,8 +8,8 @@ PROJECT(${MYNAME} C CXX)
 FILE(GLOB SRC include/*.h source/*.cpp )
 
 INCLUDE_DIRECTORIES( include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include )
 
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} )
 
diff --git a/source/lib/PccLibDecoder/CMakeLists.txt b/source/lib/PccLibDecoder/CMakeLists.txt
index b6e35fc..03fd0be 100644
--- a/source/lib/PccLibDecoder/CMakeLists.txt
+++ b/source/lib/PccLibDecoder/CMakeLists.txt
@@ -5,16 +5,16 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 PROJECT(${MYNAME} C CXX)
 
 FILE(GLOB SRC include/*.h source/*.cpp
-              ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.hpp
-              ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.h )
+              ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.hpp
+              ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.h )
 
 INCLUDE_DIRECTORIES( include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include  
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibVideoDecoder/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/nanoflann
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibColorConverter/include  )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibVideoDecoder/include
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/dependencies/nanoflann
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibColorConverter/include  )
                      
 SET( LIBS PccLibCommon PccLibBitstreamCommon PccLibVideoDecoder PccLibColorConverter )
 
diff --git a/source/lib/PccLibEncoder/CMakeLists.txt b/source/lib/PccLibEncoder/CMakeLists.txt
index c8c62be..96a3ed8 100644
--- a/source/lib/PccLibEncoder/CMakeLists.txt
+++ b/source/lib/PccLibEncoder/CMakeLists.txt
@@ -5,16 +5,16 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 PROJECT(${MYNAME} C CXX)
 
 FILE(GLOB SRC include/*.h source/*.cpp 
-              ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.hpp
-              ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.h )
+              ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.hpp
+              ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.h )
 
 INCLUDE_DIRECTORIES( include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include  
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibVideoEncoder/include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibColorConverter/include/
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/nanoflann )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibVideoEncoder/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibColorConverter/include/
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/dependencies/nanoflann )
 
 SET( LIBS PccLibCommon PccLibBitstreamCommon PccLibVideoEncoder PccLibColorConverter )
 
diff --git a/source/lib/PccLibMetrics/CMakeLists.txt b/source/lib/PccLibMetrics/CMakeLists.txt
index 4cc914c..3ff511d 100644
--- a/source/lib/PccLibMetrics/CMakeLists.txt
+++ b/source/lib/PccLibMetrics/CMakeLists.txt
@@ -5,14 +5,14 @@ STRING(REPLACE " " "_" MYNAME ${MYNAME})
 PROJECT(${MYNAME} C CXX)
 
 FILE(GLOB SRC include/*.h source/*.cpp 
-              ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.hpp
-              ${CMAKE_SOURCE_DIR}/dependencies/nanoflann/*.h )
+              ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.hpp
+              ${TMC2_SOURCE_DIR}/dependencies/nanoflann/*.h )
 
 INCLUDE_DIRECTORIES( include
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include  
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include 
-                     ${CMAKE_SOURCE_DIR}/dependencies/tbb/include
-                     ${CMAKE_SOURCE_DIR}/dependencies/nanoflann )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include
+                     ${TMC2_SOURCE_DIR}/dependencies/tbb/include
+                     ${TMC2_SOURCE_DIR}/dependencies/nanoflann )
 
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} )
 
diff --git a/source/lib/PccLibVideoDecoder/CMakeLists.txt b/source/lib/PccLibVideoDecoder/CMakeLists.txt
index 37b6a78..985bccb 100644
--- a/source/lib/PccLibVideoDecoder/CMakeLists.txt
+++ b/source/lib/PccLibVideoDecoder/CMakeLists.txt
@@ -9,8 +9,8 @@ FILE(GLOB SRC  include/*.h
                ${HM_LIB_SOURCE_DIR}/TAppCommon/* )
 
 INCLUDE_DIRECTORIES( include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include/
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include/ )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include/
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include/ )
 
 SET( LIBS PccLibCommon )
 IF( USE_HMLIB_VIDEO_CODEC )
@@ -33,15 +33,15 @@ IF( USE_VTMLIB_VIDEO_CODEC )
     SET( LIBS ${LIBS} CommonLib DecoderLib_vtm )
 ENDIF()
 IF( USE_HMAPP_VIDEO_CODEC )
-  INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/dependencies/PccLibHevcParser/include )
+  INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/dependencies/PccLibHevcParser/include )
   SET( LIBS ${LIBS} PccLibHevcParser )
 ENDIF()
 IF( USE_JMAPP_VIDEO_CODEC )
-  INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/dependencies/PccLibAvcParser/include )
+  INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/dependencies/PccLibAvcParser/include )
   SET( LIBS ${LIBS} PccLibAvcParser )
 ENDIF()
 IF( USE_SHMAPP_VIDEO_CODEC )
-  INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/dependencies/PccLibShvcParser/include )
+  INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/dependencies/PccLibShvcParser/include )
   SET( LIBS ${LIBS} PccLibShvcParser )
 ENDIF()
 ADD_LIBRARY( ${MYNAME} ${LINKER} ${SRC} )
diff --git a/source/lib/PccLibVideoEncoder/CMakeLists.txt b/source/lib/PccLibVideoEncoder/CMakeLists.txt
index efd097c..ecf789a 100644
--- a/source/lib/PccLibVideoEncoder/CMakeLists.txt
+++ b/source/lib/PccLibVideoEncoder/CMakeLists.txt
@@ -10,8 +10,8 @@ FILE(GLOB SRC include/*.h
 
 SET( LIBS PccLibCommon )
 INCLUDE_DIRECTORIES( include 
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibCommon/include/
-                     ${CMAKE_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include/ )
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibCommon/include/
+                     ${TMC2_SOURCE_DIR}/source/lib/PccLibBitstreamCommon/include/ )
 
 IF( USE_HMLIB_VIDEO_CODEC )
   INCLUDE_DIRECTORIES( ${HM_LIB_SOURCE_DIR}/ )
@@ -27,7 +27,7 @@ IF( USE_FFMPEG_VIDEO_CODEC )
   IF( WIN32 OR MSVC OR MSYS OR MINGW )
     LINK_DIRECTORIES( ${FFMPEG_DEV_DIR}/lib ) 
   ENDIF()
-  INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/source/lib/PccLibVideoDecoder/include/ )
+  INCLUDE_DIRECTORIES( ${TMC2_SOURCE_DIR}/source/lib/PccLibVideoDecoder/include/ )
   SET( LIBS ${LIBS} ${AVCODEC_LIBRARY} ${AVUTIL_LIBRARY} ${AVFORMAT_LIBRARY} PccLibVideoDecoder )    
 ENDIF()
 IF( USE_VTMLIB_VIDEO_CODEC )
