diff --git a/tmc3/PCCPointSet.h b/tmc3/PCCPointSet.h
index 1e4b390..34d78c2 100644
--- a/tmc3/PCCPointSet.h
+++ b/tmc3/PCCPointSet.h
@@ -411,7 +411,7 @@ public:
   }
 
   size_t getPointCount() const { return positions.size(); }
-  void resize(const size_t size)
+  virtual void resize(const size_t size)
   {
     positions.resize(size);
     if (hasColors()) {
@@ -428,7 +428,7 @@ public:
     }
   }
 
-  void reserve(const size_t size)
+  virtual void reserve(const size_t size)
   {
     positions.reserve(size);
     if (hasColors()) {
@@ -444,7 +444,7 @@ public:
       laserAngles.reserve(size);
     }
   }
-  void clear()
+  virtual void clear()
   {
     positions.clear();
     colors.clear();
@@ -499,7 +499,7 @@ public:
         std::next(laserAngles.begin(), dstEnd));
   }
 
-  void swapPoints(const size_t index1, const size_t index2)
+  virtual void swapPoints(const size_t index1, const size_t index2)
   {
     assert(index1 < getPointCount());
     assert(index2 < getPointCount());
